core/bind/core_bind.cpp

```diff
@@ -1673,8 +1673,14 @@ int _Directory::get_current_drive() {
 }
 
 Error _Directory::change_dir(String p_dir) {
-	ERR_FAIL_COND_V_MSG(!is_open(), ERR_UNCONFIGURED, "Directory must be opened before use.");
```

Comment:
I think the methods which don't check `is_open()` should keep checking that `d` is valid as they used to. Otherwise if the code ever changes to make `d` invalid, this is a null pointer dereference waiting to happen.